### ISSUE TITLE
NamedTuple infers empty in logic operator

### ISSUE THREAD

**Issue by iyume** [microsoft/pyright#4069]
**Describe the bug**
NamedTuple instance that has more than one field should be non-empty value.
Just like `Tuple[int]` always True, and `Tuple[()]` always False.

**To Reproduce**
```python
from typing import NamedTuple, Tuple


class A(NamedTuple):
    a: int


x: A = A(1) or None  # errors: Expression of type "A | None" cannot be assigned to declared type "A"
y: Tuple[int] = (1,) or None
```

**Comment by erictraut** [comment]
This requires some special-casing for NamedTuples and other subclasses of tuple, but I think it's a reasonable enhancement request. Mypy appears to have this logic in place.

This will be included in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.277, which I just published. It will also be included in a future release of pylance.
